### PR TITLE
HPCC-32238 Add large_list data type to Parquet Plugin

### PR DESCRIPTION
**Pull request by ilhan2316**
## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [x] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: plugins/parquet/parquetembed.cpp [hunk 1]
```diff
@@ -844,9 +844,6 @@ void ParquetWriter::beginSet(const RtlFieldInfo *field)
     arrow::ArrayBuilder *childBuilder;
     arrow::FieldPath match = getNestedFieldBuilder(field, childBuilder);
     fieldBuilderStack.push_back(std::make_shared<ArrayBuilderTracker>(field, childBuilder, CPNTSet, std::move(match)));
-
-    arrow::ListBuilder *listBuilder = static_cast<arrow::ListBuilder *>(childBuilder);
```
Comment (jackdelv) [inline plugins/parquet/parquetembed.cpp:848]: This piece of code pads the list builder and sets it up for adding list elements to it. I believe what was meant to go here was a change from a ListBuilder to LargeListBuilder.

File: plugins/parquet/parquetembed.cpp [hunk 2]
```diff
@@ -1452,8 +1449,7 @@ void ParquetRowBuilder::getDecimalResult(const RtlFieldInfo *field, Decimal &val
  * @param field Field with information about the context of the set.
  * @param isAll Not Supported.
  */
-void ParquetRowBuilder::processBeginSet(const RtlFieldInfo *field, bool &isAll)
-{
+void ParquetRowBuilder::processBeginSet(const RtlFieldInfo *field, bool &isAll) {
```
Comment (jackdelv) [inline plugins/parquet/parquetembed.cpp:1452]: Style: Curly braces should be on a line of their own.

File: plugins/parquet/parquetembed.cpp [hunk 3]
```diff
@@ -1463,12 +1459,17 @@ void ParquetRowBuilder::processBeginSet(const RtlFieldInfo *field, bool &isAll)
...
+        ParquetColumnTracker newPathNode(field, arrayVisitor->largeListArr, CPNTSet);
+        newPathNode.childCount = arrayVisitor->largeListArr->value_slice(currentRow)->length();
+        pathStack.push_back(newPathNode);
+    }
     else
     {
         failx("Error reading nested set with name %s.", field->name);
     }
 }
-
```
Comment (jackdelv) [inline plugins/parquet/parquetembed.cpp:1471]: Style: not sure why this line was removed.

File: plugins/parquet/parquetembed.hpp [hunk 4]
```diff
@@ -702,4 +710,4 @@ class ParquetEmbedFunctionContext : public CInterfaceOf<IEmbedFunctionContext>
     std::shared_ptr<ParquetWriter> parquetWriter = nullptr;     // Parquet File Writer
 };
 }
-#endif
+#endif
```
Comment (jackdelv) [inline plugins/parquet/parquetembed.hpp:713]: Style: Missing newline and end of file.

File: plugins/parquet/parquetembed.cpp (hunk 1 above)
Comment (jackdelv) [reply plugins/parquet/parquetembed.cpp:848]: Make sure that you are testing writing Parquet files with an ECL SET as this change would cause an error.

**Review by jackdelv — CHANGES_REQUESTED** [review]
@ilhan2316 I have just a few small comments.

File: plugins/parquet/parquetembed.cpp [hunk 5]
```diff
@@ -845,8 +845,8 @@ void ParquetWriter::beginSet(const RtlFieldInfo *field)
     arrow::FieldPath match = getNestedFieldBuilder(field, childBuilder);
     fieldBuilderStack.push_back(std::make_shared<ArrayBuilderTracker>(field, childBuilder, CPNTSet, std::move(match)));
 
-    arrow::ListBuilder *listBuilder = static_cast<arrow::ListBuilder *>(childBuilder);
-    reportIfFailure(listBuilder->Append());
+    arrow::LargeListBuilder *LargeListBuilder = static_cast<arrow::LargeListBuilder *>(childBuilder);
+    reportIfFailure(LargeListBuilder->Append());
```
Comment (jackdelv) [inline plugins/parquet/parquetembed.cpp:849]: Should be largeListBuilder.

**Review by jackdelv — CHANGES_REQUESTED** [review]
@ilhan2316 It's close. Just one minor comment.

**Review by jackdelv — APPROVED** [review]
@ilhan2316 The changes look good. Please squash the commits.

**Comment by jackdelv** [comment]
@dcamper Please take a look. This changes the default when writing Parquet files to use the Large String type and adds it to the checked types when reading Parquet files.

**Review by dcamper — APPROVED** [review]
Looks good to me.

**Comment by ilhan2316** [comment]
Squashed and ready to be merged. Please have a look @ghalliday

**Pull request merged**